src/Draco.Compiler/Internal/Declarations/ClassDeclaration.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Draco.Compiler.Api.Syntax;
+using Draco.Compiler.Internal.Utilities;
+
+namespace Draco.Compiler.Internal.Declarations;
+
+/// <summary>
+/// A class declaration.
+/// </summary>
+internal sealed class ClassDeclaration(ClassDeclarationSyntax syntax) : Declaration(syntax.Name.Text)
+{
+    /// <summary>
+    /// The syntax of the declaration.
+    /// </summary>
+    public ClassDeclarationSyntax Syntax { get; } = syntax;
+
+    public override ImmutableArray<Declaration> Children =>
+        InterlockedUtils.InitializeDefault(ref this.children, this.BuildChildren);
+    private ImmutableArray<Declaration> children;
+
+    public override IEnumerable<SyntaxNode> DeclaringSyntaxes
+    {
+        get
+        {
+            yield return this.Syntax;
+        }
+    }
```

Comment:
We might want to change this `DeclaringSyntaxes` to use the new collection expression in each derived class possible to make it concise.